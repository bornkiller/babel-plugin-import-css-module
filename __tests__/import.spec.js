/**
 * @description - import css module
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

const path = require('path');
const babel = require('babel-core');
const ImportCSSModule = require('../lib');

// fixture
const babelOptions = {
  presets: [],
  plugins: [],
  babelrc: false,
};

describe('babel-plugin-import-css-module', () => {
  it('should insert json', () => {
    const fixture = path.resolve(__dirname, '__fixture__', 'mascot.js');
    const Plugins = [ImportCSSModule];
    const specBabelOptions = Object.assign({}, babelOptions, {
      plugins: Plugins,
    });
    const { code } = babel.transformFileSync(fixture, specBabelOptions);

    expect(code).toMatchSnapshot();
  });

  it('should accept match rule override', () => {
    const fixture = path.resolve(__dirname, '__fixture__', 'mascot.js');
    const Plugins = [[ImportCSSModule, { rule: '\\.pcss$' }]];
    const specBabelOptions = Object.assign({}, babelOptions, {
      plugins: Plugins,
    });
    const { code } = babel.transformFileSync(fixture, specBabelOptions);

    expect(code).toMatchSnapshot();
  });
});
