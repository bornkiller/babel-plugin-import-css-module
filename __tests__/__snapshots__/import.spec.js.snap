// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-import-css-module should accept match rule override 1`] = `
"import { liberty } from './mascot.css';
import './mascot.pcss';
import { love } from './mascot.pcss.json';"
`;

exports[`babel-plugin-import-css-module should insert json 1`] = `
"import './mascot.css';
import { liberty } from './mascot.css.json';
import './mascot.pcss';
import { love } from './mascot.pcss.json';"
`;
